core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingTransformer.java

```diff
@@ -218,7 +222,6 @@ public CompletableFuture<ResponseT> prepare() {
         public void onResponse(ResponseT response) {
             if (onResponseCalled.compareAndSet(false, true)) {
                 log.trace(() -> "calling onResponse on the upstream transformer");
-                // todo: should we send back the first response to the upstreamResponseTransformer as-is?
```

Comment:
Did we decide to not do this ?

Reply:
That question was part of the surface area review. In v2, the existing CRT client does return the first response, and in v1 the multipart is done all in transfer-manager so this behaviour does not apply, the GetObjectResponse is never returned to the user. Therefore, returning the first response seems appropriate.